autopush/websocket.py

```diff
@@ -1173,12 +1234,36 @@ def ver_filter(notif):
                           message_size=size, uaid_hash=self.ps.uaid_hash,
                           user_agent=self.ps.user_agent, code=code,
                           **self.ps.raw_agent)
-            d = self.force_retry(self.ps.message.delete_message, msg)
-            # We don't remove the update until we know the delete ran
-            # This is because we don't use range queries on dynamodb and we
-            # need to make sure this notification is deleted from the db before
-            # we query it again (to avoid dupes).
-            d.addBoth(self._handle_webpush_update_remove, chid, msg)
+
+            if msg.sortkey_timestamp:
+                # Is this the last un-acked message we're waiting for?
+                last_unacked = sum(
+                    len(self.ps.updates_sent[x]) for x in self.ps.updates_sent
```

Comment:
this is simply iterating through updates_sent values so it should be an itervalues() call